File: Python/Data Structures/stack.py
from typing import Any, Boolean


class Stack:
    def __init__(self) -> None:
        """Creates a stack data structure

        Methods:
        push, pop, display, isEmpty
        """
        self.stck = []
    
    def __str__(self) -> str:
        return str(self.stck)
    
    def push(self, element) -> None:
        """Add an element to the top of the stack

        Args:
            element (<Any>): Object to be added to the top of the stack
        """
        self.stck.append(element)
    
    def pop(self) -> Any:
        """Remove an element from the top of the stack

        Raises:
            Exception: Stack Underflow

        Returns:
            <Any>: Top element of the stack that is popped
        """
        if not self.isEmpty():
            top = self.stck[-1]
            self.stck = self.stck[:-1]
            return top
        else:
            raise Exception("Stack Underflow")
    
    def isEmpty(self) -> Boolean:
        """Checks if the stack is empty or not

        Returns:
            Bool: True if the stack is empty, False otherwise
        """
        return True if self.stck == [] else False
    
    def display(self) -> None:
        """Print the entire stack as a string
        """
        print(self.stck)

File: MLSA Event-101/Data structure/Cpp/Stack/stack.cpp
// Stack --> add or remove only at one end LIFO 
// operATION --> 1)push(x) 2)pop() 3)top() 4)empty()
#include <iostream>
using namespace std;
#define n 100
class stack
{
    int* arr;
    int top;
    public:
    stack()
    {
        arr=new int [n];
        top=-1;
    }

    void push(int x)
    {
        if(top==n-1)
        {
            cout<<"stack overflow";
            return;
        }
        top++;
        arr[top]=x;
    }

    void pop()
    {
        if(top==-1)
        {
            cout<<"No element to delete"<<endl;
            return;
        }
        top--;
    }

    int Top()
    {
        if(top==-1)
        {
            cout<<"No element in Stack"<<endl;
            return -1;
        }
        return arr[top];
    }

    bool empty()
    {
        return top==-1;
    }

};
int main()
{
    stack st;
    st.push(8);
    st.push(2);
    st.push(3);
    st.push(4);
    cout<<st.Top()<<endl;
    st.pop();
    cout<<st.Top()<<endl;
    st.pop();
    st.pop();
    
    cout<<st.empty()<<endl;
    
    return 0;
}